test/SILOptimizer/definite_init_failable_initializers_diagnostics.swift

```diff
@@ -128,18 +128,17 @@ class ErrantClass : ErrantBaseClass {
     do {
       try self.init()
     } catch let e {
-      something(x) // expected-error {{'self' used inside 'catch' block reachable from self.init call}}
-      something(self.x) // expected-error {{'self' used inside 'catch' block reachable from self.init call}}
+      something(x) // expected-error* {{'self' used before 'self.init'}}
+      something(self.x) // expected-error* {{'self' used before 'self.init'}}
```

Comment:
There's special handling for `catch` blocks trying to access partially-initialized class instances, but it looks irrelevant when there either is an object or isn't. This is again because convenience initializers now get checked exactly the same as other delegating initializers.